tests/integration_test.rs

```diff
@@ -62,156 +62,213 @@ fn test_general_usage_output() {
+                );
+                assert_contains_match!(
+                    ctx.pack_stdout,
+                    "! Skipping `wget` because `wget@1.21.2-.*` is already installed on the system \\(consider removing `wget` from your project.toml configuration for this buildpack\\)"
                 );
-                assert_contains!(ctx.pack_stdout, "Adding xmlsec1@1.2.33-1build2");
-                assert_contains_match!(ctx.pack_stdout, "! Skipping wget because wget@1.21.2-.* is already installed on the system \\(consider removing wget from your project.toml configuration for this buildpack\\)");
                 assert_contains!(
                     ctx.pack_stdout,
-                    "! Virtual package libvips is provided by libvips42@8.12.1-1build1 (consider replacing libvips for libvips42 in your project.toml configuration for this buildpack)"
+                    "! Virtual package `libvips` is provided by `libvips42@8.12.1-1build1` (consider replacing `libvips` with `libvips42` in your project.toml configuration for this buildpack)"
                 );
                 assert_contains!(
                     ctx.pack_stdout,
-                    "! Skipping libvips42 because libvips42@8.12.1-1build1 is already installed on the system (consider removing libvips42 from your project.toml configuration for this buildpack)"
+                    "! Skipping `libvips42` because `libvips42@8.12.1-1build1` is already installed on the system (consider removing `libvips42` from your project.toml configuration for this buildpack)"
                 );
 
                 assert_contains!(ctx.pack_stdout, "## Installing packages");
-
-                assert_contains!(ctx.pack_stdout, "Downloading and extracting libgwenhywfar79");
-                assert_contains!(ctx.pack_stdout, "Downloading and extracting xmlsec1");
-                assert_contains!(ctx.pack_stdout, "Downloading and extracting libgwenhywfar-data");
+                assert_contains!(ctx.pack_stdout, "Requesting packages");
+                assert_contains!(ctx.pack_stdout, "`libgwenhywfar79@5.9.0-1` from http://archive.ubuntu.com/ubuntu/pool/universe/libg/libgwenhywfar/libgwenhywfar79_5.9.0-1_amd64.deb");
+                assert_contains!(ctx.pack_stdout, "`libgwenhywfar-data@5.9.0-1` from http://archive.ubuntu.com/ubuntu/pool/universe/libg/libgwenhywfar/libgwenhywfar-data_5.9.0-1_all.deb");
+                assert_contains!(ctx.pack_stdout, "`xmlsec1@1.2.33-1build2` from http://archive.ubuntu.com/ubuntu/pool/main/x/xmlsec1/xmlsec1_1.2.33-1build2_amd64.deb");
+                assert_contains!(ctx.pack_stdout, "Downloading");
+                assert_contains!(ctx.pack_stdout, "Installation complete");
+                assert_contains!(ctx.pack_stdout, "Layer file listing");
+                assert_contains!(ctx.pack_stdout, "/layers/heroku_deb-packages/packages/usr/bin/xmlsec1");
```

Comment:
Do we need these lines? I think in most cases, this will just be an annoying thing to scroll/skip when looking for language buildpack details. If users care about what files are installed, they can figure it out pretty easily by inspecting the resulting image (or use a `heroku run` shell).

Reply:
I think it is beneficial to provide a record of exactly what was installed. It can be useful for:
- debugging purposes 
- when you need to provide some extra configuration and want to know where exactly your packages were installed to (e.g.; setting jemalloc in an environment variable like in https://github.com/heroku/heroku-buildpack-apt/issues/131)

See also:
- https://github.com/heroku/heroku-buildpack-apt/issues/130
- https://github.com/heroku/heroku-buildpack-apt/issues/96

Reply:
I still feel like text like this is spammy:

```
      /layers/heroku_deb-packages/packages/usr/bin/xmlsec1
      /layers/heroku_deb-packages/packages/usr/share/locale/de/LC_MESSAGES/gwenhywfar.mo
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/struct_base.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_bindata.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/tree_base.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/uint16_t.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_multicache_type.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/uint16_t_array.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_param_list2.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_xmlnode_list2.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/uint32_t.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/int_base.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/int.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_buffer.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/char_ptr.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_multicache.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/char.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/list2_base.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/uint32_t_array.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/double_array.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_stringlist2.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/uint64_t.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/uint8_t.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_idlist64.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/char_array.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/double.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_date.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_xmlnode_list.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/int_array.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_param_tree.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_xmlnode.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/list1_base.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_param_list.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_param.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_db_node.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/tree2_base.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/idmap_base.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/gwen_time.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/typemaker2/c/uint8_t_array.tm2
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/dialogs/dlg_message.dlg
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/dialogs/dlg_progress.dlg
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/dialogs/dlg_showbox.dlg
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/dialogs/dlg_input.dlg
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/tm2builder.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/staticlib.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/tmplib.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/cbuilder.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/sharedlib.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/app.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/cxxapp.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/cxxbuilder.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/module.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/posix/msgfmt.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/tm2builder.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/staticlib.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/tmplib.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/cbuilder.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/sharedlib.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/app.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/cxxapp.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/cxxbuilder.gwb
      /layers/heroku_deb-packages/packages/usr/share/gwenhywfar/gwenbuild/builders/windows/module.gwb
      /layers/heroku_deb-packages/packages/usr/share/doc/libgwenhywfar-data/copyright
      /layers/heroku_deb-packages/packages/usr/share/doc/libgwenhywfar-data/changelog.Debian.gz
      /layers/heroku_deb-packages/packages/usr/share/doc/libgwenhywfar79t64/copyright
      /layers/heroku_deb-packages/packages/usr/share/doc/libgwenhywfar79t64/changelog.Debian.gz
      /layers/heroku_deb-packages/packages/usr/share/doc/xmlsec1/copyright
      /layers/heroku_deb-packages/packages/usr/share/man/man1/xmlsec1.1.gz
      /layers/heroku_deb-packages/packages/usr/share/gwenbuild/templates/project.tmpl
      /layers/heroku_deb-packages/packages/usr/share/gwenbuild/templates/module.tmpl
      /layers/heroku_deb-packages/packages/usr/share/gwenbuild/templates/convlib.tmpl
      /layers/heroku_deb-packages/packages/usr/share/lintian/overrides/libgwenhywfar79t64
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/libgwengui-cpp.so.79.10.2
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/libgwenhywfar.so.79.10.2
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/configmgr/dir.xml
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/configmgr/dir.so
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/ct/ohbci.xml
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/ct/ohbci.so
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/dbio/olddb.xml
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/dbio/olddb.so.0.0.1
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/dbio/csv.xml
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/dbio/xmldb.xml
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/dbio/xmldb.so.0.0.1
      /layers/heroku_deb-packages/packages/usr/lib/x86_64-linux-gnu/gwenhywfar/plugins/79/dbio/csv.so.0.0.1

Reply:
@joshwlewis what if we put it behind a flag? I can see this output as helpful in some but cases but maybe not all cases. It could be off by default.

Reply:
I updated my comment above with a different suggestion (I always forget whether I need to cmd+enter or just enter =P).

A flag that disables the file listing by default also makes sense to me. 

I'm also happy if you want to file either idea away as an issue to tackle some other time.

Reply:
## Flag thoughts

There's a standard log level env you can use from paketo https://github.com/paketo-buildpacks/rfcs/blob/11691011512b9f075d06c31b74ea29bb5c3e5f69/text/0027-log-levels.md. I think it makes more sense to adopt that standard than to invent our own.

## Debug verbose thoughts

I can't find a link, but I've also proposed the concept of a debug log file in lieu of flags or log levels. The idea is that: When people want more information, it's often because something unexpected happened. That can mean an intermittent problem that's difficult to reproduce. Or even if it's consistent, but the build takes a long time it means a long delay until people get feedback. The (theoretical) feature would allow buildpack maintainers to always write debug information for every build, and make it available for later consumption. I didn't go as far as writing an RFC though as I'm not entirely sure how to implement it, and it wasn't a urgent need. I got generally positive feedback from the paketo and buildpacks team on my vague sketch though. 

If the info we're wanting to log isn't likely to cause a build failure then we could implement a trial version ourselves by storing the contents in the layer, or maybe making a build only layer that stores it (and then we need to figure out how to emit it later) if we don't want it visible at runtime.

Reply:
Using `BP_LOG_LEVEL` environment variable to control the verbosity introduced by the file listing. It will now only be shown when `BP_LOG_LEVEL=DEBUG`.